src/utils/test/uri.ts

```diff
@@ -1,50 +1,90 @@
-import { Erc1319URI } from 'ethpm/utils/uri';
+import { EthpmURI } from 'ethpm/utils/uri';
+import { URL } from 'url';
 
 describe('validates URIs', () => {
   const validRegistryUris = [
+    'erc1319://snakecharmers.eth',
     'erc1319://snakecharmers.eth:1',
     'erc1319://snakecharmers.eth:1/',
+    new URL('erc1319://snakecharmers.eth:1/'),
```

Comment:
Are `url` objects only used internally? Are users expected to/given the option to use them? Just curious/context-gathering, but will note that while flexibility can be nice, adoption can be aided by presenting one clear, "best" way to do the thing.